DIPs/1NNN-ADR.md

```diff
@@ -0,0 +1,479 @@
+
+### Diagnostics
+
+An `i""` string is likely not be compatible with existing D `string` functions and variables. The compiler SHOULD provide helpful hints in the error messages to point new users in the right direction to understand how to use the new feature. This message MAY link to a web page to educate the user on which solution is appropriate for them.
+
+```
+void foo(string s) {}
+
+foo(i"test $(4)"); // a CT error due to type mismatch. The compiler should suggest functions (see below) to remedy the user's problem.
+```
+
+### Concatenations
+
+In order to facilitate convenient formatting of long strings, if a `StringLiteral` follows an `InterpolatedString`,
+it is appended to the `InterpolatedString` in the parsing pass. (Not the lexer pass.)
+A `StringPostfix` may not appear on any of them.
+Such concatenations enable the various string literal types to be used as `InterpolatedStrings`,
+such as:
+
+```
+i""q{apples and $("bananas")}
+```
+yielding a tuple expression:
+```
+_d_interpolated_string!("apples and ", _d_interpolated_format_spec(null)), ("bananas")
+```
+
+### Library additions
+
+While the compiler only lowers to library calls and thus does not need to know about any of this, user functionality relies on a small library implementation.
+
+One suggested implementation would be:
+
+```
+struct _d_interpolated_string(Parts...) {
+        static:
+
+        private bool hasAllSpecs() {
+            foreach(part; Parts)
+               static if(is(typeof(part) == _d_interpolated_format_spec))
+                  if(part.spec is null)
+                       return false;
+            return true;
+        }
+
+        private string toFormatStringImpl(string defaultSpec) {
+           if(__ctfe) {
+                string ret;
+                foreach(part; Parts)
+                        static if(is(typeof(part) == _d_interpolated_format_spec)) {
+                                if(part.spec is null)
+                                        ret ~= defaultSpec;
+                                else
+                                        ret ~= part.spec;
+                        } else static if(is(typeof(part) : const(char)[]))
+                                ret ~= part;
+                        else static assert(0);
+                return ret;
+           } else assert(0);
+        }
+
+        private immutable(char*) toFormatStringzImpl() {
+                return toFormatString!(null).ptr;
+        }
+
+	// we may also add a lambda for escaping chars to this
+        public template toFormatString(string defaultSpec) {
+                enum toFormatString = toFormatStringImpl(defaultSpec);
+        }
+
+        static if(hasAllSpecs()) {
+                // alias this to a string, even a string literal, loses
+                // the implicit conversion to const(char)* needed by
+                // printf. so gotta use a wrapper that explicitly returns
+                // that :(
```

Comment:
There should be a bugzilla on this. There's zero reason for the compiler to lose this implicit conversion. Note that it works fine if the enum is done in some cases